fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/runtime/MetricsRegistryImpl.java

```diff
@@ -286,13 +286,15 @@ public void removeMatching(MetricFilter metricFilter) {
     }
 
     private <T extends Metric> SortedMap<String, T> getMetrics(MetricType type, MetricFilter filter) {
-        SortedMap<String, T> out = new TreeMap<String, T>();
+        SortedMap<String, T> out = new TreeMap<>();
 
-        Iterator<Map.Entry<String, Metric>> iterator = metricMap.entrySet().iterator();
-        while (iterator.hasNext()) {
-            Map.Entry<String, Metric> entry = iterator.next();
+        for (Map.Entry<String, Metric> entry : metricMap.entrySet()) {
             if (filter.matches(entry.getKey(), entry.getValue())) {
-                out.put(entry.getKey(), (T) entry.getValue());
+                Metadata metadata = metadataMap.get(entry.getKey());
+                assert metadata != null : "missing metadata for registered metric";
```

Comment:
So if metadata are missing and assertions are disabled the next line would just throw NPE?

Reply:
See https://github.com/Ladicek/wildfly-swarm/blob/5ac9db2cee25adc0a44d51e753e40458692dd2a8/fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/runtime/MetricsRegistryImpl.java#L82-L84